conf/find_invalid_waivers.py

```diff
@@ -0,0 +1,161 @@
+    else:
+        # tuple with the regexes not found, add it to the list
+        regexes_matched_list.append((regexes, non_pass_result))
+
+
+def unwaive_note(waive_text, note):
+    return note.lstrip(waive_text).rstrip(')')
+
+
+def match_result_mark_waiver(regexes_matched_list, version, arch, status, name, note):
+    """This function is an updated version of the match_result() function from the lib/waive.py."""
+    version = versions._Rhel(version=version, id='rhel')
+
+    # make sure "'something' in name" always works
+    if name is None:
+        name = ''
+    if note is None or note == '[]':
+        note = ''
+
+    if '(waived pass)' in note:
+        note = unwaive_note('(waived pass)', note)
+    elif '(waived fail)' in note:
+        note = unwaive_note('(waived fail)', note)
+        status = 'fail'
+    elif '(waived error)' in note:
+        note = unwaive_note('(waived error)', note)
+        status = 'error'
+
+    objs = {
+        'status': status,
+        'name': name,
+        'note': note,
+        'arch': arch,
+        'rhel': version,
+        'oscap': '',
+        'env': '',
+        'Match': waive.Match,
+    }
+
+    for section in _sections_cache:
+        if any(x.fullmatch(name) for x in section.regexes):
+            ret = eval(section.python_code, objs, None)
+
+            if not isinstance(ret, waive.Match):
+                if not isinstance(ret, bool):
+                    raise RuntimeError(f"waiver python code did not return bool or Match: {ret}")
+                ret = waive.Match(ret)
+
+            if ret:  # both regex and python code matched
+                add_regexes_to_list(regexes_matched_list, section.regexes, bool(status != 'pass'))
+
+
+def load_and_process_results(file, regexes_matched_list):
+    with gzip.open(file, 'rt') as f:
+        for line in f:
+            line = line.strip()
+            version, arch, status, name, note = line.split('\t')
+
+            if status not in ['pass', 'fail', 'error', 'warn']:
+                continue
+
+            match_result_mark_waiver(regexes_matched_list, version, arch, status, name, note)
+
+
+def main(result_file_list):
+    regexes_matched_list = []
+    """
+    [
+        # list of tuples containing regexes matching some test result and a Boolean value
+        # set to True if regexes matched any non-pass test result, otherwise False
+        (set(regexes), True/False),
+        (set(regexes), True/False),
+        ...
+    ]
+    """
+    global _sections_cache
+    _sections_cache = list(waive.collect_waivers())
+
+    for file in result_file_list:
+        load_and_process_results(file, regexes_matched_list)
+
+    print("===============================================================\n"
+          "The following waivers are no longer valid, they either did not\n"
+          "match any test results or only matched the 'pass' test results:\n"
+          "===============================================================\n")
```

Comment:
This is something we hit with Jan as well.

The problem with split strings (`"..."` separated by spaces or newlines, NOT `,`-separated arguments) is that flake8/Ruff treats them as a **single argument**, which they technically are.

This means that
```python
print(
    "foo "
    "bar"
)
```
triggers an error, because it sees a multi-line function call, but doesn't see `,` on each line, which cannot be there, because it's not multiple `print` arguments, it's one split argument.

The counter-intuitive, but probably most correct, solution is to add one `,` one the last line of the split string, satisfying the condition that a multi-like function call must have `,`-terminated arguments.

```suggestion
    print(
        "===============================================================\n"
        "The following waivers are no longer valid, they either did not\n"
        "match any test results or only matched the 'pass' test results:\n"
        "===============================================================\n",
    )
```
This makes it consistent with other multi-line function calls.